src/main/java/core/basesyntax/strategy/impl/PurchaseOperationProcessor.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.model.Transaction;
+import core.basesyntax.strategy.OperationProcessor;
+
+public class PurchaseOperationProcessor implements OperationProcessor {
+    private final FruitDao fruitDao = new FruitDaoImpl();
```

Comment:
```suggestion
    private final FruitDao fruitDao;
```
pass the instance of the dao through the constructor